consensus/types/src/light_client_update.rs

```diff
@@ -120,22 +119,67 @@ impl<T: EthSpec> LightClientUpdate<T> {
         // Build finalized header from finalized block
         let finalized_header = BeaconBlockHeader {
             slot: finalized_block.slot(),
-            proposer_index: finalized_block.proposer_index(),
+            proposer_index: finalized_block.message().proposer_index(),
             parent_root: finalized_block.parent_root(),
             state_root: finalized_block.state_root(),
-            body_root: finalized_block.body_root(),
+            body_root: finalized_block.message().body_root(),
         };
         if finalized_header.tree_hash_root() != beacon_state.finalized_checkpoint().root {
             return Err(Error::InvalidFinalizedBlock);
         }
         let next_sync_committee_branch =
             attested_state.compute_merkle_proof(NEXT_SYNC_COMMITTEE_INDEX)?;
         let finality_branch = attested_state.compute_merkle_proof(FINALIZED_ROOT_INDEX)?;
+
+        if let Some(deneb_fork_epoch) = chain_spec.deneb_fork_epoch {
+            if beacon_state.slot().epoch(T::slots_per_epoch()) >= deneb_fork_epoch {
+                return Ok(Self {
+                    attested_header: LightClientHeaderDeneb::block_to_light_client_header(
+                        attested_block,
+                    )?
+                    .into(),
+                    next_sync_committee: attested_state.next_sync_committee()?.clone(),
+                    next_sync_committee_branch: FixedVector::new(next_sync_committee_branch)?,
+                    finalized_header: LightClientHeaderDeneb::block_to_light_client_header(
+                        finalized_block,
+                    )?
+                    .into(),
+                    finality_branch: FixedVector::new(finality_branch)?,
+                    sync_aggregate: sync_aggregate.clone(),
+                    signature_slot: block.slot(),
+                });
+            }
+        };
+
+        if let Some(capella_fork_epoch) = chain_spec.capella_fork_epoch {
+            if beacon_state.slot().epoch(T::slots_per_epoch()) >= capella_fork_epoch {
+                return Ok(Self {
+                    attested_header: LightClientHeaderCapella::block_to_light_client_header(
+                        attested_block,
+                    )?
+                    .into(),
+                    next_sync_committee: attested_state.next_sync_committee()?.clone(),
+                    next_sync_committee_branch: FixedVector::new(next_sync_committee_branch)?,
+                    finalized_header: LightClientHeaderCapella::block_to_light_client_header(
+                        finalized_block,
+                    )?
+                    .into(),
+                    finality_branch: FixedVector::new(finality_branch)?,
+                    sync_aggregate: sync_aggregate.clone(),
+                    signature_slot: block.slot(),
+                });
+            }
+        };
+
```

Comment:
I would turn that into a match to the Fork enum, and return a tuple (attested_header, finalized_header)